### ISSUE TITLE
Upgrade problem: Profile images mapped to wrong picture

### ISSUE THREAD

**Issue by rfay** [warmshowers/Warmshowers.org#729]
In the most recent fresh upgrade (2015-09-02) I see that several thousand user profile images are mapped inappropriately to wrong images (like recommendation photos). 

For example, user/1, user/15497, should be able to look for users with picture that is a recommendation_photo. 2648 of them are wrong, as seen by this query: 

`select u.uid,f.fid,f.uri from users u, file_managed f where u.picture = f.fid and f.uri NOT LIKE "public://pictures/%";`

or

`select u.uid,f.fid,f.uri from users u, file_managed f where u.picture = f.fid and f.uri NOT LIKE CONCAT("public://pictures/picture-", u.uid, "%");`

Good news: 35,843 of them are (apparently) right.

This probably means it's likely that other pictures are also mapped wrong.

**Comment by rfay** [comment]
I suspect my fix in 180e8846fec24e7669f0cf3bcd5bd700c8bc899b - That looks like a sloppy replace(), and should be reverted and reviewed during the upgrade.

**Comment by rfay** [comment]
This is a "thing". 

- [Migrating image fields ended up swapping files to userpic](https://www.drupal.org/node/1558192)

It looks like [#1732944: Corrupted file URI after filefield migration](https://www.drupal.org/node/1732944) is probably relevant to us and important. I don't know if it can solve this problem.

There are a couple of other file-related anomalies. One is that I'm doing this upgrade on a case-insensitive filesystem, which following the discussion in [D6 -> D7 upgrade: Duplicate files cause Integrity constraint violation: 1062 Duplicate entry 'xx' for key 'PRIMARY'](https://www.drupal.org/node/1176186#comment-4550470) could lead to some trouble. It would be tragic if my problem were just that I'm doing this on a case-insensitive filesystem.

**Comment by rfay** [comment]
I debugged it to find what was actually happening, and the full description is [in the issue](https://www.drupal.org/node/1558192#comment-10327715) - it turns out to be a very serious content_migrate/cck data corruption problem. The fix has been added into the [updb instructions](https://github.com/warmshowers/Warmshowers.org/issues/531#issuecomment-113533347). It's ugly - Comment out a section of user.install during the core upgrade, do everything through the field migration, reset the commented-out section, run update.php again.
